lib/cfa/augeas_parser.rb

```diff
@@ -186,29 +212,23 @@ def load_from_augeas(aug, prefix, keys_cache)
       @data = keys_cache.keys_for_prefix(prefix).map do |key|
         aug_key = prefix + "/" + key
         {
-          key:   load_key(prefix, aug_key),
-          value: load_value(aug, aug_key, keys_cache)
+          key:       load_key(prefix, aug_key),
+          value:     load_value(aug, aug_key, keys_cache),
+          orig_key:  stripped_path(prefix, aug_key),
+          operation: :keep
         }
       end
     end
 
-    # @note for internal usage only
-    # @api private
-    #
-    # Saves {#data} to *prefix* in *aug*.
-    # @param aug [::Augeas]
-    # @param prefix [String] Augeas path prefix
-    # @return [void]
-    def save_to_augeas(aug, prefix)
-      arrays = {}
-
-      @data.each do |entry|
-        save_entry(entry[:key], entry[:value], arrays, aug, prefix)
+    def ==(other)
+      return false if self.class != other.class
+      other_data = other.data # do not compute again
```

Comment:
well, as it is just Array, I find it too dark magic to overwrite Array default `==`